### PR TITLE
Plasma Cutter Change

### PR DESCRIPTION
**Pull request by Arctrooper8**
## About The Pull Request

Changes the plasma cutter to no longer allow for the cell to be removed, the starting cell itself is replaced with a recharging cell (Values subject to change, feedback highly appreciated). 

(And I must stress that these are open for discussion)
Current variables 
PC cost: 20 point (Unchanged)
PC Cell max capacity: 8500 power
PC Cell charge delay: 2 seconds (Same as the other recharging cells)
PC Cell charge amount: 25 power (How much is charged every 2 seconds)

Deconstruction costs (not changed by this PR)
Normal wall: 1000 power (Has a use time)
Resin wall: 200 power overall (Has no use time but you have to hit the wall two times at 100 cost per hit)

## Why It's Good For The Game

Right now balancing the PC is based around one major aspect, power logistics. For example, there are three major ways to handle the power logistics I've noticed:
1. Carrying a bag of cells with you, this allows you to simply replace the cell whenever you need to and enables you to speed through not only resin defenses but more normal walls too (vast empty shooting fields).
2. Taking the hand charger, although you will face some down time the hand charger is incredibly strong and can effectively charge the PC's starting cell rapidly allowing for liberal application of the PC on resin and normal walls.
3. Buying a self recharging cell from the engi vend and letting it work on it's own, pales in comparison to the other except for it's ease of usage.
Of course you can combine any of the above methods for greater effect.

This is a lot to account for when balancing one tool. By forcing the PC to have a single non-removable cell the PC become much easier to balance, and slightly easier to use.

Of course this does raise some other questions:
Hand crank is now only useful for lasgun and APC cells (MJP wants it removed but that's another PR)
Recharger cells are probably never going to be purchased anymore on their own (I'd be interested to see if anybody used them for something other than PC though)

Overall though this makes the PC easier to use while making it vastly simpler to balance.

## Changelog
:cl:
balance: PC cell can no longer be removed but is now self recharging.
/:cl:

### CODE REVIEW

**Comment by Derrick2Electricboogaloo** [comment]
Ew, 200 per hit from 100. So now engineers get the PC for more, so they have less points overall, kinda not worth. Also MJP plans on killing hand charger when he does his engineering kit rework. I would rather just stick with swapping my batteries rather than waiting for it to slowly charge.

**Comment by Arctrooper8** [comment]
> Ew, 200 per hit from 100. So now engineers get the PC for more, so they have less points overall, kinda not worth. Also MJP plans on killing hand charger when he does his engineering kit rework. I would rather just stick with swapping my batteries rather than waiting for it to slowly charge.

Removing hand charger is a good idea but it will result in more people carrying around absurd amounts of cells and deleting swathes of the maps. The benefit of forcing a cell is that you can more easily control balance.
And that 200 per hit from 100 might be a mistake on my part, I could've misread the comments which talked about supposed power per hit cost
And all the variables from price to charge rate are discussable, I won't pretend to be some genius balancer.

**Comment by Arctrooper8** [comment]
>Also MJP plans on killing hand charger when he does his engineering kit rework.

Hahahahahahahahahahahahahahahaha
Soon TM
MJP said if I code it I can do the rework

**Comment by Arctrooper8** [comment]
>Also MJP plans on killing hand charger when he does his engineering kit rework.

Hahahahahahahahahahahahahahahaha
Soon TM
MJP said if I code it I can do the rework

**Comment by Arctrooper8** [comment]
So the earlier resin stuff was because I was hitting the walls too quickly when testing, so I assumed it was broken. In reality resin walls take a while to reach their full health it seems. Regardless that misguided cost change is gone.

**Comment by EvilMelonTheSecond** [comment]
If we are gonna nerf it so heavily maybe make it fit into bags or something. As it stands this is practically shadowrealming it.

**Comment by Arctrooper8** [comment]
> If we are gonna nerf it so heavily maybe make it fit into bags or something. As it stands this is practically shadowrealming it.

The major point of this PR is to make the PC easier to balance. Cell bags and hand cranking generally allow someone to invalidate most xeno resin defenses if not the entire map. Take a look at these:

>(And I must stress that these are open for discussion)
Current variables
PC cost: 30 points (Was 20)
PC Cell max capacity: 3000 power
PC Cell charge delay: 2 seconds (Same as the other recharging cells)
PC Cell charge amount: 25 power (How much is charged every 2 seconds)

Those are the variables and they're extremely open to discussion, you can suggest new ones.

**Comment by Derrick2Electricboogaloo** [comment]
That is fine, I am not sure why we are so worried about a finite amount of high cap cells when that requires someone to put in the effort and kit sacrificing other things. Sure you can put backpacks somewhere in a backline, but that means you have to get off the front and move away which is time not killing resin.

**Comment by goose4429** [comment]
The resin wall balance seems fine to me in this PR. You basically have 15 charges, and regain a charge once in 16 seconds. Marines probably won't like the cost increase, but the PC is underrated.

**Comment by Losenis** [comment]
> 25 recharge per 2 seconds
> 200 to kill resin
> 18 seconds per resin
> 90 seconds per shittyass wall
> Nerfs capacity by 70 fucking percent for an item that WILL get dropped at the first knockdown and acided

Are you out of your fucking mind

**Comment by Arctrooper8** [comment]
> > 25 recharge per 2 seconds
> > 200 to kill resin
> > 18 seconds per resin
> > 90 seconds per shittyass wall
> > Nerfs capacity by 70 fucking percent for an item that WILL get dropped at the first knockdown and acided
> 
> Are you out of your fucking mind

I've said it again and again, the variables aren't final and I'd appreciate suggestions instead of just whining.


The current numbers are based off the already existing recharging cells.
The major point of this PR is to make the PC easier to balance, specifically against the oppressive power of carrying a backpack full of cells or using the hand crank. The PC should be good at clearing resin, not good at deleting half the map.

And if it being acided is a problem maybe it could have some sort of mag harness system.

**Comment by InterroLouis** [comment]
The price increase seems pretty unnecessary considering the massive nerfs the plasma cutter is getting.

**Comment by Starfirejordan** [comment]
nerf unwarranted, shouldn't happen all together.

**Comment by Arctrooper8** [comment]
> The price increase seems pretty unnecessary considering the massive nerfs the plasma cutter is getting.

My god, could it be? A helpful suggestion?
Excuse my sarcasm but thank you for actually suggesting something instead of just seething.
I shall revert the price increase

**Comment by chesse20** [comment]
this seems incredibly annoying, I already had to spend 20 points on the plasmacutter, 13 points on powercells, and fill up two backpack slots, an entire pouch, and my entire webbing to be an effective plasmacutter main, and now I'm going to have to spend 60 points on 3 plasmacutters, juggle between them like a clown, and give up my backpack slot so I can carry one plasmacutter in my belt, one in my hand, and one on my back.

**Comment by EvilMelonTheSecond** [comment]
I like the idea of making PC actually balancable instead of getting rid of it or making it scarce. My suggestion would be to make clearing resin walls almost free since xeno can and will spam them to stop marine advances. As for normal wall clearing I got no clue, we are probably better off doing trial and error since nobody knows if it should even be allowed

**Comment by breadhunt** [comment]
V Text wall below V



Max capacity is definitely worth upping. Charge rate is kinda bad, but 3k max is _awful_. I'd go as far as saying the old 10k is fine. ~~or 20k~~ 
3k is **three walls** you can cut after saving up for, like, 10 minutes. Or **15 FUCKING RESIN WALLS**. Whatthefuck that's pitiful and you can't even tunnel properly. 
Higher capacity lets the engi go do other stuff while the cutter recharges instead of having to babysit the plasmacutter. Then return like a badass and go do cool cutting shit.

Magharn PC is cool but being droppable gives xenos a way to counter the cutter.
Making it fit on suit storage doesn't prevent melting, but it does mean if you need belt+back slot, you can wear PC on suit storage and hold the gun. Which means you don't need to dedicate your build as much around PC


I think this is a really good PR. Is it a nerf? Yeah. But with finetuning during the TM it might not be a nerf to the ground. 

You cannot say clicking walls for 2 hours straight is fun. Even if you have to handcrank and replace a cell every like 5 minutes (riveting gameplay)
This turns PC into more of a cool side thing you go do every 10-15 minutes. Plus the entire build isn't around cramming like 20 cells in anymore, so there's room for mats, or a teleporter, or a _gun_, or whatever. More options in a build good

TL;DR up max capacity to 10k+ and add PC to suitstorage so engis can go do other stuff. good PR

**Comment by Losenis** [comment]
>This turns PC into more of a cool side thing you go do every 10-15 minutes. Plus the entire build isn't around cramming like 20 cells in anymore, so there's room for mats, or a teleporter, or a gun, or whatever. More options in a build good

What kind of reasoning is this? It's not making it a cool thing to do every 10-15 minutes, it's making it something you can ONLY do every 10-15 minutes because in its current iteration you can do it almost continuously if you wanted to due to cell swapping, but with this proposal you're only going to remove a few walls and then be locked out of it for a while. And you already have the option as engi to get a recharger cell for the PC instead of normal cells, now this is outright forcing you down that path in a downgraded version removing all other options. In fact you even had the option of going for both and getting a recharger cell and then some normal cells.

You can already do what you're imagining. The proposed pr would just be locking you down that path without deviation, so you'd have less options overall.

**Comment by breadhunt** [comment]
> You can already do what you're imagining. 

Yeah you're right, you can already do it, and it's _fucking awesome_. A couple were saying that not being able to switch cells kills PC or something, which I really disagree with. Also higher cap leans into it more
I did phrase it badly with "this turns PC into x", sorry. I'm saying PC still has a fuckton of build options as rechargable

> you can do it almost continuously

That's the main (only?) balance issue with plasmacutter, and that's why this is a nerf
The ability to do it continuously is insanely strong, unfun for both sides and constantly complained about. All three of those are pretty unhealthy for the game if nothing else. Added bonus is PC probably won't get trashed in the process and is still useful.

> you're only going to remove a few walls 

If you mean metal walls, that's the intention. Turns out xenos literally need cover to live, and open space + flares denies every single caste except boiler in the game. Flanking and tunneling is still possible. If you really need to clear a space, like around an FOB, you can still clear it, it'll just take.. some time. Or an OB.
If you mean resin walls, numbers are up to be changed, and unless Arc is literally insane, they will be, because a single resin wall every 24 seconds is kinda bad

> and then be locked out of it for a while

This is good because it:
a) gives xenos breaks between having to focus on defending the maze/taking out PC guy/watching out for flanks
b) the above gives xenos opportunity to counterattack, or for marines to conserve PC charge for their own counterattacks/pushes
c) gives engis a break from fighting, which while (as you said) can already be done, often isn't because of how powerful cell switching is. The _only_ bad thing I can see about this is:
> The proposed pr would just be locking you down that path without deviation, so you'd have less options overall.

which, yeah, you're right. That's the downside of the PR. I think that it's a net gain though, considering how awful for the game PC is right now. I've seen like two engis taking PC+rechargables though, just because of cell swapping being so fucking powerful if you're taking PC, you might as well not do anything else. Razing an entire map to deny any sort of cover from offscreen shit is fair and balanced.

**Comment by Arctrooper8** [comment]
Breadhunt basically covered everything, thanks Breadhunt.
The voices in my head say that I am mentally very well and advised me to make changes based on feedback.

But first, some math:
The cell charges at 25 power every 2 seconds
That means every 16 seconds we can gain enough charge to kill a resin wall
And every 80 seconds we gain enough charge to kill a metal wall/rock

**2000 size cell** (Small recharging cell)
160 seconds to fully charge from zero
worth ten resin walls
**3000 size cell** (Current value)
240 seconds to fully charge from zero
worth 15 resin walls
**7500**
600 seconds to fully charge from zero
worth 37.5 resin walls
**8500** (New value)
640 seconds to fully charge from zero
worth 42.5 resin walls
**9500**
720 seconds to fully charge from zero
worth 47.5 resin walls

**New commit changes:**
The PC can now fit into suit storage
The cell size is now 8500

**Review by BraveMole — APPROVED** [review]
Seems interesting enough to be TM. I agree that it didn't need a price change

**Comment by Naaanii** [comment]
Yeah, these changes are fairly decent. Doesn't really stop PC engineers from being oppressive nor does it change the fact that an engineer can change terrain to completely favour marines. The downtime allows for Xenos to retaliate a bit more, but even then, any form of cover or support fire will allow engineers to freely alter the map, albeit at a slower pace.

Resin wall nerf isn't much but it'd be better if PC were more used on resin walls rather than actual walls or terrain, but the current stats seem alright. Recharger cells will need an overhaul however as they seem like a complete waste of points for the moment.

With price remaining the same, don't really see this as a heavy nerf and this doesn't really stop PC from being oppressive to Xenos either and is just a minor nerf overall, trading logistics for downtime, but is a better alternative than hard nerfing or even removing it.

**Comment by Alexeivichi** [comment]
I was promised more alcohol in this PR so this has my support for a speedmerge.

**Comment by EvilMelonTheSecond** [comment]
The charge amount seems brutally low even for resin clearing. For a single wall you gotta wait 80 seconds and not clear any resin while at it. Maybe increase the charge amount and normal wall cutting cost so it keeps the resin clearing utility?
